src/components/ApplicationLink/ApplicationLink.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import {Link} from 'react-router-dom';
+
+interface ApplicationLinkProps {
+  classes: string;
+  title: JSX.Element;
+  url: string;
+}
+
+const ApplicationLink: React.SFC<ApplicationLinkProps> = ({
+  classes,
+  title,
+  url
+}) => {
+  if (url === window.location.href) {
+    return (
+      <a className={classes} key={url}>
+        {title}
+      </a>
+    );
+  }
+  if (!url.includes('http')) {
+    return (
+      <Link to={url} className={classes} key={url}>
+        {title}
+      </Link>
+    );
+  }
+  return (
```

Comment:
This block is very similar to the first one. I think it make sense to check for HTTP only, then prepare href link and put it into the a link